compiler/il/OMRDataTypes.cpp

```diff
@@ -92,31 +87,16 @@ OMR::DataType::getMaxPrecisionFromType()
 TR::DataType
 OMR::DataType::getVectorIntegralType()
    {
-   switch(self()->getDataType())
-      {
-      case TR::VectorInt8:
-      case TR::VectorInt16:
-      case TR::VectorInt32:
-      case TR::VectorInt64: return self()->getDataType();
-      case TR::VectorFloat: return TR::VectorInt32;
-      case TR::VectorDouble: return TR::VectorInt64;
-      default:
-         return TR::NoType;
-         break;
-      }
-   }
+   if (!self()->isVector()) return TR::NoType;
 
-TR::DataType
-OMR::DataType::getVectorElementType()
-   {
-   switch(self()->getDataType())
+   switch(self()->getVectorElementType())
       {
-      case TR::VectorInt8: return TR::Int8;
-      case TR::VectorInt16: return TR::Int16;
-      case TR::VectorInt32: return TR::Int32;
-      case TR::VectorInt64: return TR::Int64;
-      case TR::VectorFloat: return TR::Float;
-      case TR::VectorDouble: return TR::Double;
+      case TR::Int8:
+      case TR::Int16:
+      case TR::Int32:
+      case TR::Int64: return self()->getDataType();
+      case TR::Float: return TR::DataType::createVectorType(TR::Int32, self()->getVectorLength());
+      case TR::Double: return TR::DataType::createVectorType(TR::Int64, self()->getVectorLength());
```

Comment:
I am not sure who uses this method but that's what it did originally: returned integral type of the same element size as the input type.

Reply:
If all vector element types are equally valid then this seems wrong to me.  At the very least it should be understood why we can't simply return "self()->getDataType()" like the other element types rather than accepting that what's there now is correct.